test/parallel/test-fs-rm.js

```diff
@@ -181,6 +232,39 @@ if (isGitPresent) {
     fs.rmSync(filePath, common.mustNotMutateObjectDeep({ force: true }));
   }
 
+  // Should delete a valid symlink
+  const linkTarget = path.join(tmpdir.path, 'link-target.txt');
+  fs.writeFileSync(linkTarget, '');
+  const validLink = path.join(tmpdir.path, 'valid-link');
+  fs.symlinkSync(linkTarget, validLink);
+  try {
+    fs.rmSync(validLink);
+  } finally {
+    fs.rmSync(linkTarget, common.mustNotMutateObjectDeep({ force: true }));
+    fs.rmSync(validLink, common.mustNotMutateObjectDeep({ force: true }));
+  }
+
+  // Should delete an invalid symlink
+  const invalidLink = path.join(tmpdir.path, 'invalid-link');
+  fs.symlinkSync('definitely-does-not-exist', invalidLink);
+  try {
+    fs.rmSync(invalidLink);
+  } finally {
+    fs.rmSync(invalidLink, common.mustNotMutateObjectDeep({ force: true }));
+  }
+
+  // Should delete a symlink that is part of a loop
+  const loopLinkA = path.join(tmpdir.path, 'loop-link-a');
+  const loopLinkB = path.join(tmpdir.path, 'loop-link-b');
+  fs.symlinkSync(loopLinkA, loopLinkB);
+  fs.symlinkSync(loopLinkB, loopLinkA);
+  try {
+    fs.rmSync(loopLinkA);
```

Comment:
Same here, and optionally can be fixed in old code
```suggestion
    fs.rmSync(loopLinkA);
    assert.strictEqual(fs.existsSync(loopLinkA), false);
```

Reply:
Thanks, done.

I also fixed in existing code even though it's not strictly related to the issue this PR is fixing. If that's not okay I can open a separate PR for these changes instead, no problem.

I also just want to note that this would probably not have happened if this would've been shared code between the callback, promise and sync version tests, because someone clearly thought of this for the callback version but forgot to make the same change in two other places.

Reply:
Thank you. I think, fixing those in this PR is fine.

Historically, `fs` tests tend to be repetitive because Promises API came much later than others, and usually combining sync/awaited code with Callback API in one wrapper function is not trivial. Many tests are even split into 3 separate files to avoid complexity and potential race conditions.
In general (e.g. in new tests) I agree with your suggested approach; but IMHO refactoring this particular test might make it less readable.

Reply:
Alright, thanks for the explanation.